watertap/core/membrane_channel1d.py

```diff
@@ -0,0 +1,295 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+from pyomo.common.config import Bool, ConfigValue, In
+from pyomo.environ import (
+    Constraint,
+    NonNegativeReals,
+    NegativeReals,
+    Param,
+    Set,
+    Var,
+    value,
+    units as pyunits,
+)
+from idaes.core import (
+    declare_process_block_class,
+    DistributedVars,
+    FlowDirection,
+    useDefault,
+)
+from idaes.core.base.control_volume1d import ControlVolume1DBlockData
+from idaes.core.util import scaling as iscale
+from idaes.core.util.misc import add_object_reference
+import idaes.logger as idaeslog
+
+from watertap.core.membrane_channel_base import (
+    MembraneChannelMixin,
+    PressureChangeType,
+    CONFIG_Template as Base_CONFIG_Template,
+)
+
+CONFIG_Template = Base_CONFIG_Template()
+
+CONFIG_Template.declare(
+    "area_definition",
+    ConfigValue(
+        default=DistributedVars.uniform,
+        domain=In(DistributedVars),
+        description="Argument for defining form of area variable",
+        doc="""Argument defining whether area variable should be spatially
+variant or not. **default** - DistributedVars.uniform.
+**Valid values:** {
+DistributedVars.uniform - area does not vary across spatial domain,
+DistributedVars.variant - area can vary over the domain and is indexed
+by time and space.}""",
+    ),
+)
```

Comment:
This is also a core IDAES feature - this makes me wonder if this should be importing a standard config block from somewhere in IDAES, as it looks like it is re-creating its own.

Also, do not feel that you need to duplicate all config argument from IDAES - there are far more than you will need for any one case, and you should only implement those that you need. The underlying framework will default to the most general case if no input is given.

Reply:
I think it's a hold-over from 1DRO (which this is copied from) which wasn't used. Will eliminate.

Reply:
I just want to note that this was added to RO1D since the area can technically vary across the spatial domain. Although `DistributedVars.variant` wasn't being used, I included this for a potential future application, if we ever want to consider variable area across the length of the membrane channel.

Reply:
@adam-a-a would like to maintain this, so I reverted the change that removed it. I don't have strong feelings either way.